app/models/post.rb

```diff
@@ -0,0 +1,18 @@
+require_relative 'comment'
+
+class Post < ApplicationRecord
+  after_initialize :update_posts_counter
+  has_many :comments
+  belongs_to :author, class_name: 'User'
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts_counter.to_i + 1)
+  end
```

Comment:
- [ OPTIONAL] I recommend keeping the update likes method private since you used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, so I think it would be better if you keep as you make It first 💪🏻✅👏🏻💯🔥